### ISSUE TITLE
Typescript Errors when compiling with vite

### ISSUE THREAD

**Issue by mseele** [SortableJS/vue.draggable.next#41]
When i use vue.draggable.next i got typescript errors on compile time (errors are printed at the end).

### Github Repo to reproduce
https://github.com/mseele/vue.draggable.next-ts-issue

### Step by step scenario
- Checkout  https://github.com/mseele/vue.draggable.next-ts-issue
- Run `yarn run build`
- See the errors

**OR**

- run `yarn create @vitejs/app vue.draggable.next-ts-issue --template vue-ts`
- Replace `App.vue` content with https://github.com/SortableJS/vue.draggable.next/blob/master/example/components/transition-example-2.vue
- replace `<script>` with `<script lang="ts">`
- replace `export default {` with `export default defineComponent({`
- add `import { defineComponent } from "vue";`
- Run `yarn run build`
- See the errors

These are the errors i get:
```
node_modules/vuedraggable/src/vuedraggable.d.ts:2:17 - error TS2305: Module '"vue"' has no exported member 'VueConstructor'.

2   import Vue, { VueConstructor } from 'vue';
                  ~~~~~~~~~~~~~~

node_modules/vuedraggable/src/vuedraggable.d.ts:5:22 - error TS2709: Cannot use namespace 'Vue' as a type. 

5     Instance extends Vue,
                       ~~~

node_modules/vuedraggable/src/vuedraggable.d.ts:13:22 - error TS2709: Cannot use namespace 'Vue' as a type.

13     Instance extends Vue,
                        ~~~

node_modules/vuedraggable/src/vuedraggable.d.ts:19:69 - error TS2709: Cannot use namespace 'Vue' as a type.

19     CombinedVueInstance<Instance, Data, Methods, Computed, Props> & Vue
                                                                       ~~~

node_modules/vuedraggable/src/vuedraggable.d.ts:30:16 - error TS2709: Cannot use namespace 'Vue' as a type.

30     component: Vue;
                  ~~~

node_modules/vuedraggable/src/vuedraggable.d.ts:58:5 - error TS2709: Cannot use namespace 'Vue' as a type.

58     Vue,
```

**Comment by dicarne** [comment]
Same problem!
A hack method may help:
modify `node_modules\vuedraggable\src\vuedraggable.d.ts` to
```
declare module 'vuedraggable' {
  //import Vue, { VueConstructor } from 'vue';
//
  //type CombinedVueInstance<
  //  Instance extends Vue,
  //  Data,
  //  Methods,
  //  Computed,
  //  Props
  //> = Data & Methods & Computed & Props & Instance;
//
  //type ExtendedVue<
  //  Instance extends Vue,
  //  Data,
  //  Methods,
  //  Computed,
  //  Props
  //> = VueConstructor<
  //  CombinedVueInstance<Instance, Data, Methods, Computed, Props> & Vue
  //>;
//
  //export type DraggedContext<T> = {
  //  index: number;
  //  futureIndex: number;
  //  element: T;
  //};
//
  //export type DropContext<T> = {
  //  index: number;
  //  component: Vue;
  //  element: T;
  //};
//
  //export type Rectangle = {
  //  top: number;
  //  right: number;
  //  bottom: number;
  //  left: number;
  //  width: number;
  //  height: number;
  //};
//
  //export type MoveEvent<T> = {
  //  originalEvent: DragEvent;
  //  dragged: Element;
  //  draggedContext: DraggedContext<T>;
  //  draggedRect: Rectangle;
  //  related: Element;
  //  relatedContext: DropContext<T>;
  //  relatedRect: Rectangle;
  //  from: Element;
  //  to: Element;
  //  willInsertAfter: boolean;
  //  isTrusted: boolean;
  //};

  const draggable: any;
  //ExtendedVue<
  //  Vue,
  //  {},
  //  {},
  //  {},
  //  {
  //    options: any;
  //    list: any[];
  //    value: any[];
  //    clone: any;
  //    tag?: string | null;
  //    move: any;
  //    componentData: any;
  //  }
  //>;

  export default draggable;
}

```

Just ignore any type def. I build success after that.

**Comment by saibbyweb** [comment]
Facing the same issue. Also encountered the same error pattern while using another package `(vue3-rouch-events)`. The dev server works fine, but builds fail.

**Comment by saibbyweb** [comment]
> Facing the same issue. Also encountered the same error pattern while using another package `(vue3-rouch-events)`. The dev server works fine, but builds fail.

I was able to bypass build errors by adding:
` "skipLibCheck": true` to my tsconfig.json's `compilerOptions` property.

**Comment by Tumselass** [comment]
`"skipLibCheck": true` did not work for me. 
Commenting out the declaration file seems to be the easiest shortcut for now. 

Am I right to assume that the declaration file looks more at home with Vue 2 and needs an update to vue 3?

**Comment by David-Desmaisons** [comment]
Corrected in version 4.1.0
